### PR TITLE
pilot: fix `types=wds` not working in `/debug/config_dump`

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

```
istioctl x internal-debug "config_dump?types=wds&proxyID=sleep-668646db99-85x2m.sidecar"
```

### CODE REVIEW

File: pilot/pkg/xds/debug.go [hunk 1]
```diff
@@ -698,6 +704,9 @@ func (s *DiscoveryServer) connectionConfigDump(conn *Connection, includeEds bool
 		v3.SecretType,
 		v3.EndpointType,
 		v3.ExtensionConfigurationType,
+		v3.WorkloadType,
+		v3.AddressType,
+		v3.WorkloadAuthorizationType,
```
Comment (howardjohn) [inline pilot/pkg/xds/debug.go:709]: why do we need to change the default? It is meant to mirror envoy's

**Review by howardjohn — COMMENTED** [review]
fwiw there is also `ambientz` and `krtz` which give basically the same. But this looks good too

File: pilot/pkg/xds/debug.go (hunk 1 above)
Comment (zirain) [reply pilot/pkg/xds/debug.go:709]: I understand `no types parameter` mean dump all configurations.

Comment (howardjohn) [reply pilot/pkg/xds/debug.go:709]: I don't think so, its already
```
	// EDS is disabled by default for compatibility with Envoy config_dump interface
	if includeEds {
```

for example. This interface is used for `istioctl ps` to diff envoy vs pilot

Comment (zirain) [reply pilot/pkg/xds/debug.go]: make sense, reverted.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**